lib/app-common.ts

```diff
@@ -15,6 +15,36 @@
  * limitations under the License.
  */
 
+class CliSettings {
+	public readonly settings: any;
+	constructor() {
+		// TODO figure out why the typescript compiler attempts to type-check
+		// the `balena-settings-client` module (and then fails with errors) if
+		// a straighforward `require('balena-settings-client')` statement is
+		// used here. It may even be a compiler bug, because `tsconfig.json`
+		// has a `"skipLibCheck": true` setting.
+		this.settings = require(['balena', 'settings', 'client'].join('-'));
```

Comment:
This was baffling. See comment above. I only created the `CliSettings` class to isolate this oddity and avoid spreading these `require/join` lines in multiple places in the file.
* `require('balena-settings-client')` fails on type checking the `balena-settings-client` module, which is a CLI dependency and should not be type checked.
* `require(['balena', 'settings', 'client'].join('-'))` works, presumably because the typescript compiler does not attempt to execute the join at compilation time in order to obtain the module name.